--- a/paramiko/transport.py	2022-08-18 07:24:07.000000000 -0700
+++ b/paramiko/transport.py	2022-08-18 07:24:17.000000000 -0700
@@ -232,12 +232,6 @@
             "block-size": 16,
             "key-size": 32,
         },
-        "blowfish-cbc": {
-            "class": algorithms.Blowfish,
-            "mode": modes.CBC,
-            "block-size": 8,
-            "key-size": 16,
-        },
         "aes128-cbc": {
             "class": algorithms.AES,
             "mode": modes.CBC,
