--- a/setup.py	2023-02-27 02:42:19.000000000 -0800
+++ b/setup.py	2023-02-27 02:43:14.000000000 -0800
@@ -65,6 +65,7 @@
             "Quartz.CoreVideo._CVPixelBuffer",
             ["Modules/_CVPixelBuffer.m"],
             extra_link_args=["-framework", "Quartz"],
+            extra_compile_args=["-Wno-error=unused-const-variable"],
             py_limited_api=True,
         ),
         # CoreGraphics
