diff --git a/pyobjc-framework-libdispatch/pyobjc_setup.py b/pyobjc-framework-libdispatch/pyobjc_setup.py
index 34f2d34..b135272 100644
--- a/pyobjc-framework-libdispatch/pyobjc_setup.py
+++ b/pyobjc-framework-libdispatch/pyobjc_setup.py
@@ -514,11 +514,22 @@ def Extension(*args, **kwds):
     # XCode 15 has a bug w.r.t. weak linking for older macOS versions,
     # fall back to older linker when using that compiler.
     # XXX: This should be in _fixup_compiler but doesn't work there...
-    lines = subprocess.check_output(["xcodebuild", "-version"], text=True).splitlines()
-    if lines[0].startswith("Xcode"):
-        xcode_vers = int(lines[0].split()[-1].split(".")[0])
-        if xcode_vers >= 15:
-            ldflags.append("-Wl,-ld_classic")
+    xcode_vers = None
+    try:
+        lines = subprocess.check_output(
+            ["xcodebuild", "-version"], text=True
+        ).splitlines()
+        if lines[0].startswith("Xcode"):
+            xcode_vers = int(lines[0].split()[-1].split(".")[0])
+    except subprocess.CalledProcessError:
+        lines = subprocess.check_output(
+            ["pkgutil", "--pkg-info=com.apple.pkg.CLTools_Executables"],
+            text=True).splitlines()
+        version, = [l.split()[1] for l in lines
+            if l.startswith('version: ')]
+        xcode_vers = int(version.split(".")[0])
+    if xcode_vers >= 15:
+        ldflags.append("-Wl,-ld_classic")
 
     if os_level == "10.4":
         cflags.append("-DNO_OBJC2_RUNTIME")
