diff --git a/rencode/_rencode.pyx b/rencode/_rencode.pyx
index 5716c2d..14e1bec 100644
--- a/rencode/rencode.pyx	2021-08-05 21:15:32.117793791 +0700
+++ b/rencode/rencode.pyx	2021-08-05 21:15:29.931809176 +0700
@@ -527,6 +527,9 @@
         return decode_fixed_dict(data, pos)
     elif typecode == CHR_DICT:
         return decode_dict(data, pos)
+    else:
+        raise Exception("invalid typecode %i" % typecode)
+
 
 def loads(data, decode_utf8=False):
     """
